Flaky Test Build&Test / filebeat-goIntegTest / github.com/elastic/beats/v7/filebeat/input/filestream.TestFilestreamEmptyLine

## Flaky Test

* **Test Name:** TestFilestreamEmptyLine
* **Link:** https://github.com/elastic/beats/blob/68e990993fc5b6ce9e360befc2bef53713aaf8c0/filebeat/input/filestream/input_integration_test.go#L198
* **Branch:** master
* **Artifact Link:** https://beats-ci.elastic.co/job/Beats/job/beats/job/PR-27572/3/testReport/junit/github/com_elastic_beats_v7_filebeat_input_filestream/Build_Test___filebeat_goIntegTest___TestFilestreamEmptyLine/
https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats/detail/PR-27572/3/pipeline/


### Stack Trace

```
=== RUN   TestFilestreamEmptyLine
    environment_test.go:188: 
        	Error Trace:	environment_test.go:188
        	            				input_integration_test.go:229
        	Error:      	Not equal: 
        	            	expected: 58
        	            	actual  : 37
        	Test:       	TestFilestreamEmptyLine
--- FAIL: TestFilestreamEmptyLine (1.01s)




```

--- a/filebeat/input/filestream/environment_test.go
+++ b/filebeat/input/filestream/environment_test.go
@@ -39,6 +39,7 @@ import (
 	"github.com/elastic/beats/v7/libbeat/logp"
 	"github.com/elastic/beats/v7/libbeat/statestore"
 	"github.com/elastic/beats/v7/libbeat/statestore/storetest"
+	"github.com/elastic/go-concert/unison"
 )
 
 type inputTestingEnvironment struct {
@@ -50,7 +51,8 @@ type inputTestingEnvironment struct {
 	pluginInitOnce sync.Once
 	plugin         v2.Plugin
 
-	wg sync.WaitGroup
+	wg  sync.WaitGroup
+	grp unison.TaskGroup
 }
 
 type registryEntry struct {
@@ -70,7 +72,9 @@ func newInputTestingEnvironment(t *testing.T) *inputTestingEnvironment {
 }
 
 func (e *inputTestingEnvironment) mustCreateInput(config map[string]interface{}) v2.Input {
+	e.grp = unison.TaskGroup{}
 	manager := e.getManager()
+	manager.Init(&e.grp, v2.ModeRun)
 	c := common.MustNewConfigFrom(config)
 	inp, err := manager.Create(c)
 	if err != nil {
@@ -88,12 +92,13 @@ func (e *inputTestingEnvironment) getManager() v2.InputManager {
 
 func (e *inputTestingEnvironment) startInput(ctx context.Context, inp v2.Input) {
 	e.wg.Add(1)
-	go func(wg *sync.WaitGroup) {
+	go func(wg *sync.WaitGroup, grp *unison.TaskGroup) {
 		defer wg.Done()
+		defer grp.Stop()
 
 		inputCtx := input.Context{Logger: logp.L(), Cancelation: ctx}
 		inp.Run(inputCtx, e.pipeline)
-	}(&e.wg)
+	}(&e.wg, &e.grp)
 }
 
 func (e *inputTestingEnvironment) waitUntilInputStops() {

--- a/filebeat/input/filestream/input_integration_test.go
+++ b/filebeat/input/filestream/input_integration_test.go
@@ -196,7 +196,6 @@ func TestFilestreamCloseEOF(t *testing.T) {
 
 // test_empty_lines from test_harvester.py
 func TestFilestreamEmptyLine(t *testing.T) {
-	t.Skip("Flaky test https://github.com/elastic/beats/issues/27585")
 	env := newInputTestingEnvironment(t)
 
 	testlogName := "test.log"

